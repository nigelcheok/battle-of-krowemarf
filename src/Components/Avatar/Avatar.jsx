import React from 'react';

export function Avatar(props) {
  return (
      <div className="card border-square mt-3 flex-grow-1 mr-3 shadow-on-hover" style={AvatarStyle()}>
        <a href={props.link} target="_blank" style={AnchorStyle()}>
        <div className="card-body" style={AvatarBodyStyle()}>
          <img src={props.avatarUrl} style={AvatarImageStyle()} height="60" width="60" alt={props.name}/>
          <div className="mt-2" style={AvatarTitleStyle()}>
            { props.name }
          </div>
          <div style={AvatarDescriptionStyle()}>
            { props.description }
          </div>
        </div>
        </a>
      </div>
  );
};

function AnchorStyle() {
  return {
    textDecoration: 'none'
  }
}

function AvatarStyle() {
  return {
    minWidth: '350px',
    backgroundColor: 'rgba(37,39,41,.01)',
    textAlign: 'center',
  }
}

function AvatarBodyStyle() {
  return {
    textDecoration: 'none !important',
    color: 'rgba(37,39,41)'
  }
}

function AvatarImageStyle() {
  return {
    borderRadius: '40px',
  }
}

function AvatarTitleStyle() {
  return {
    fontWeight: '600',
  }
}

function AvatarDescriptionStyle() {
  return {
    fontSize: '.8rem',
  }
}
