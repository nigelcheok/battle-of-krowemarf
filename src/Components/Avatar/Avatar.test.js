import React from 'react';
import { shallow } from 'enzyme';
import { Avatar } from './Avatar';

describe('Avatar', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<Avatar/>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with name and description', () => {
    const component = shallow(<Avatar name="test" description="test description"/>);
    expect(component).toMatchSnapshot();
  });
});