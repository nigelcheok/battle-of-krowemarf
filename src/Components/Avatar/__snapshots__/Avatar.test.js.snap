// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render correctly with name and description 1`] = `
<div
  className="card border-square mt-3 flex-grow-1 mr-3 shadow-on-hover"
  style={
    Object {
      "backgroundColor": "rgba(37,39,41,.01)",
      "minWidth": "350px",
      "textAlign": "center",
    }
  }
>
  <a
    rel="noopener noreferrer"
    style={
      Object {
        "textDecoration": "none",
      }
    }
    target="_blank"
  >
    <div
      className="card-body"
      style={
        Object {
          "color": "rgba(37,39,41)",
          "textDecoration": "none !important",
        }
      }
    >
      <img
        alt="test"
        height="60"
        style={
          Object {
            "borderRadius": "40px",
          }
        }
        width="60"
      />
      <div
        className="mt-2"
        style={
          Object {
            "fontWeight": "600",
          }
        }
      >
        test
      </div>
      <div
        style={
          Object {
            "fontSize": ".8rem",
          }
        }
      >
        test description
      </div>
    </div>
  </a>
</div>
`;

exports[`Avatar should render correctly with no props 1`] = `
<div
  className="card border-square mt-3 flex-grow-1 mr-3 shadow-on-hover"
  style={
    Object {
      "backgroundColor": "rgba(37,39,41,.01)",
      "minWidth": "350px",
      "textAlign": "center",
    }
  }
>
  <a
    rel="noopener noreferrer"
    style={
      Object {
        "textDecoration": "none",
      }
    }
    target="_blank"
  >
    <div
      className="card-body"
      style={
        Object {
          "color": "rgba(37,39,41)",
          "textDecoration": "none !important",
        }
      }
    >
      <img
        height="60"
        style={
          Object {
            "borderRadius": "40px",
          }
        }
        width="60"
      />
      <div
        className="mt-2"
        style={
          Object {
            "fontWeight": "600",
          }
        }
      />
      <div
        style={
          Object {
            "fontSize": ".8rem",
          }
        }
      />
    </div>
  </a>
</div>
`;
